src/lib/support/tests/TestUtf8.cpp

```diff
@@ -0,0 +1,170 @@
+
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include <functional>
+
+#include <lib/support/UnitTestRegistration.h>
+#include <lib/support/utf8.h>
+
+#include <nlunit-test.h>
+
+namespace {
+
+using namespace chip;
+
+bool IsValidCStringAsUtf8(const char * s)
+{
+    return Utf8::IsValid(CharSpan(s, strlen(s)));
```

Comment:
`Span::fromCharString(s)` is same as `CharSpan{s, strlen(s)}`

Reply:
I did not notice that method. Updated.